### PR TITLE
Fix weird intermittent test failure in CI

### PR DESCRIPTION
**Pull request by fubuloubu**
Because it's annoying and I hate it

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -82,18 +80,16 @@ jobs:
...
-          ~/.vvm
-        key: ${{ runner.os }}-compiler-cache
+          ~/.brownie
+          ./build
+        key: build-cache
 
     - name: Setup node.js
       uses: actions/setup-node@v1
-      with:
-        node-version: '12.x'
```
Comment (Macarse) [inline .github/workflows/test.yaml:96]: why did you remove this? I don't know what's the default, but let's have a hardcoded version at least so it's reproducible.

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -82,18 +80,16 @@ jobs:
...
-    - name: Cache compiler installations
+    - name: Cache Brownie Build Folders
       uses: actions/cache@v2
       with:
         path: |
-          ~/.solcx
-          ~/.vvm
-        key: ${{ runner.os }}-compiler-cache
+          ~/.brownie
+          ./build
```
Comment (Macarse) [inline .github/workflows/test.yaml:88]: with this, ci would compile only the diff.

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -82,18 +80,16 @@ jobs:
           ~/.vvm
         key: compiler-cache
 
-    - name: Cache compiler installations
+    - name: Cache Brownie Build Folders
       uses: actions/cache@v2
       with:
         path: |
-          ~/.solcx
-          ~/.vvm
```
Comment (Macarse) [inline .github/workflows/test.yaml:90]: why did you remove these? they make sense

File: .github/workflows/test.yaml (hunk 1 above)
Comment (fubuloubu) [reply .github/workflows/test.yaml:96]: That's not even a hardcoded version, it just uses 14.x by default

File: .github/workflows/test.yaml (hunk 2 above)
Comment (fubuloubu) [reply .github/workflows/test.yaml:88]: Yep!

File: .github/workflows/test.yaml (hunk 3 above)
Comment (fubuloubu) [reply .github/workflows/test.yaml:90]: it's in 2 places

**Review by Macarse — APPROVED** [review]
